File: scripts/create-icons.js
const fs = require('fs');
const path = require('path');

// Simple PNG generator for a blue Q icon
function createPNG(width, height) {
  // PNG file structure
  const signature = Buffer.from([0x89, 0x50, 0x4E, 0x47, 0x0D, 0x0A, 0x1A, 0x0A]);

  // IHDR chunk
  const ihdrData = Buffer.alloc(13);
  ihdrData.writeUInt32BE(width, 0);
  ihdrData.writeUInt32BE(height, 4);
  ihdrData[8] = 8;  // bit depth
  ihdrData[9] = 6;  // color type (RGBA)
  ihdrData[10] = 0; // compression
  ihdrData[11] = 0; // filter
  ihdrData[12] = 0; // interlace

  const ihdr = createChunk('IHDR', ihdrData);

  // Create image data (RGBA)
  const rawData = [];
  const centerX = width / 2;
  const centerY = height / 2;
  const radius = Math.min(width, height) / 2 - 2;

  for (let y = 0; y < height; y++) {
    rawData.push(0); // filter byte
    for (let x = 0; x < width; x++) {
      const dx = x - centerX;
      const dy = y - centerY;
      const dist = Math.sqrt(dx * dx + dy * dy);

      // Blue color #3B82F6
      const r = 59, g = 130, b = 246, a = 255;

      if (dist < radius - 3) {
        // Inside - check if part of Q
        const inQ = (x > centerX && y > centerY) ? // Q tail quadrant
          ((x - centerX) + (y - centerY) < radius / 2) :
          (Math.abs(dx) < radius / 3 && dy > -radius / 2 && dy < radius / 2); // Q body

        if (inQ) {
          rawData.push(r, g, b, a);
        } else {
          rawData.push(0, 0, 0, 0); // transparent
        }
      } else if (dist <= radius) {
        // Ring border
        rawData.push(r, g, b, a);
      } else {
        rawData.push(0, 0, 0, 0); // transparent
      }
    }
  }

  // Compress with zlib
  const zlib = require('zlib');
  const compressed = zlib.deflateSync(Buffer.from(rawData), { level: 9 });
  const idat = createChunk('IDAT', compressed);

  // IEND chunk
  const iend = createChunk('IEND', Buffer.alloc(0));

  return Buffer.concat([signature, ihdr, idat, iend]);
}

function createChunk(type, data) {
  const length = Buffer.alloc(4);
  length.writeUInt32BE(data.length, 0);

  const typeBuffer = Buffer.from(type);
  const crcData = Buffer.concat([typeBuffer, data]);
  const crc = Buffer.alloc(4);
  crc.writeUInt32BE(crc32(crcData), 0);

  return Buffer.concat([length, typeBuffer, data, crc]);
}

function crc32(data) {
  let crc = 0xFFFFFFFF;
  const table = [];

  for (let i = 0; i < 256; i++) {
    let c = i;
    for (let j = 0; j < 8; j++) {
      c = (c & 1) ? (0xEDB88320 ^ (c >>> 1)) : (c >>> 1);
    }
    table[i] = c;
  }

  for (let i = 0; i < data.length; i++) {
    crc = table[(crc ^ data[i]) & 0xFF] ^ (crc >>> 8);
  }

  return (crc ^ 0xFFFFFFFF) >>> 0;
}

const iconsDir = path.join(__dirname, '..', 'assets', 'icons');
fs.mkdirSync(iconsDir, { recursive: true });

// Create tray icon (16x16)
const tray16 = createPNG(16, 16);
fs.writeFileSync(path.join(iconsDir, 'tray-icon.png'), tray16);
console.log('Created tray-icon.png (16x16)');

// Create 256x256 icon
const icon256 = createPNG(256, 256);
fs.writeFileSync(path.join(iconsDir, 'icon.png'), icon256);
console.log('Created icon.png (256x256)');

console.log('Done!');

File: scripts/create_icons.py
#!/usr/bin/env python3
"""Generate tray icons for QuickCapture"""
import os
import struct
import zlib

def create_png(width, height, pixels):
    """Create a minimal PNG file"""
    def png_chunk(chunk_type, data):
        chunk = chunk_type + data
        crc = zlib.crc32(chunk) & 0xffffffff
        return struct.pack('>I', len(data)) + chunk + struct.pack('>I', crc)

    # PNG signature
    signature = b'\x89PNG\r\n\x1a\n'

    # IHDR chunk
    ihdr_data = struct.pack('>IIBBBBB', width, height, 8, 6, 0, 0, 0)  # 8-bit RGBA
    ihdr = png_chunk(b'IHDR', ihdr_data)

    # IDAT chunk (image data)
    raw_data = b''
    for y in range(height):
        raw_data += b'\x00'  # filter type none
        for x in range(width):
            idx = (y * width + x) * 4
            raw_data += pixels[idx:idx+4]

    compressed = zlib.compress(raw_data, 9)
    idat = png_chunk(b'IDAT', compressed)

    # IEND chunk
    iend = png_chunk(b'IEND', b'')

    return signature + ihdr + idat + iend

def create_icon(size):
    """Create a simple Q icon"""
    pixels = bytearray(size * size * 4)

    center = size // 2
    radius = size // 2 - 2

    for y in range(size):
        for x in range(size):
            idx = (y * size + x) * 4
            # Distance from center
            dx = x - center
            dy = y - center
            dist = (dx*dx + dy*dy) ** 0.5

            if dist < radius and dist > radius - 3:
                # Ring - blue color
                pixels[idx] = 59   # R
                pixels[idx+1] = 130 # G
                pixels[idx+2] = 246 # B
                pixels[idx+3] = 255 # A
            elif dist <= radius - 3 and x > center - radius//3 and y < center + radius//3:
                # Q tail - same blue
                pixels[idx] = 59
                pixels[idx+1] = 130
                pixels[idx+2] = 246
                pixels[idx+3] = 255
            else:
                # Transparent
                pixels[idx:idx+4] = [0, 0, 0, 0]

    return create_png(size, size, bytes(pixels))

def create_tray_icon(size):
    """Create a tray icon - simple Q letter"""
    pixels = bytearray(size * size * 4)
    center = size // 2
    radius = size // 2 - 1

    for y in range(size):
        for x in range(size):
            idx = (y * size + x) * 4
            dx = x - center
            dy = y - center
            dist = (dx*dx + dy*dy) ** 0.5

            # Simple filled circle with Q letter approximation
            if dist < radius - 2:
                # Inside circle - draw a simplified Q
                in_q_tail = (x > center and y > center and
                            (x - center) + (y - center) < radius // 2)
                if in_q_tail or (abs(dx) < radius//3 and abs(dy) < radius//2):
                    # Q or vertical bar
                    pixels[idx] = 59
                    pixels[idx+1] = 130
                    pixels[idx+2] = 246
                    pixels[idx+3] = 255
                else:
                    # Transparent
                    pixels[idx:idx+4] = [0, 0, 0, 0]
            elif dist <= radius:
                # Border
                pixels[idx] = 59
                pixels[idx+1] = 130
                pixels[idx+2] = 246
                pixels[idx+3] = 255
            else:
                pixels[idx:idx+4] = [0, 0, 0, 0]

    return create_png(size, size, bytes(pixels))

def main():
    icons_dir = os.path.join(os.path.dirname(__file__), '..', 'assets', 'icons')
    os.makedirs(icons_dir, exist_ok=True)

    # Create tray icons (16x16 and 32x32 for macOS retina)
    tray_16 = create_tray_icon(16)
    tray_32 = create_tray_icon(32)

    with open(os.path.join(icons_dir, 'tray-icon.png'), 'wb') as f:
        f.write(tray_16)

    # Create larger icons for app
    icon_256 = create_icon(256)

    print(f"Icons created in {icons_dir}")

if __name__ == '__main__':
    main()
